packages/apollo-datasource-rest/src/RESTDataSource.ts

```diff
@@ -74,14 +74,14 @@ export abstract class RESTDataSource<TContext = any> extends DataSource {
   protected resolveURL(request: RequestOptions): ValueOrPromise<URL> {
     let path = request.path;
     if (path.startsWith('/')) {
-      path = path.slice(1);
+      path = encodeURIComponent(path.slice(1));
```

Comment:
I don't think this makes sense. The expected use of this API is that the path is specified as a literal in the code, and multi-component paths (`this.get('foo/bar/baz')`) are common (and assuming that an API will treat `GET /foo%2Fbar%2Fbaz` equivalently to `GET /foo/bar/baz` seems incorrect).